File: 백준/Silver/10866. 덱/덱.py
from collections import deque
input = __import__('sys').stdin.readline

dq = deque()

N = int(input())
for i in range(N):
    t = list(input().split())
    if t[0] == "push_front":
        dq.appendleft(int(t[1]))
    elif t[0] == "push_back":
        dq.append(int(t[1]))
    elif t[0] == "pop_front":
        if len(dq) != 0:
            print(dq.popleft())
        else:
            print(-1)
    elif t[0] == "pop_back":
        if len(dq) != 0:
            print(dq.pop())
        else:
            print(-1)
    elif t[0] == "size":
        print(len(dq))
    elif t[0] == "empty":
        if len(dq) == 0:
            print(1)
        else:
            print(0)
    elif t[0] == "front":
        if len(dq) != 0:
            print(dq[0])
        else:
            print(-1)
    elif t[0] == "back":
        if len(dq) != 0:
            print(dq[-1])
        else:
            print(-1)

File: 백준/Silver/10866. 덱/덱.cc
#include <bits/stdc++.h>
using namespace std;

int n, x;
string str;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    deque<int> DQ;

    cin >> n;
    while (n--) {
        cin >> str;
        if (str == "push_front") {
            cin >> x;
            DQ.push_front(x);
        } else if (str == "push_back") {
            cin >> x;
            DQ.push_back(x);
        } else if (str == "pop_front") {
            if (DQ.empty()) {
                cout << -1 << '\n';
            } else {
                cout << DQ.front() << '\n';
                DQ.pop_front();
            }
        } else if (str == "pop_back") {
            if (DQ.empty()) {
                cout << -1 << '\n';
            } else {
                cout << DQ.back() << '\n';
                DQ.pop_back();
            }
        } else if (str == "size") {
            cout << DQ.size() << '\n';
        } else if (str == "empty") {
            if (DQ.empty()) {
                cout << 1 << '\n';
            } else {
                cout << 0 << '\n';
            }
        } else if (str == "front") {
            if (DQ.empty()) {
                cout << -1 << '\n';
            } else {
                cout << DQ.front() << '\n';
            }
        } else if (str == "back") {
            if (DQ.empty()) {
                cout << -1 << '\n';
            } else {
                cout << DQ.back() << '\n';
            }
        }
    }

    return 0;
}